demonstrations/qnspsa.py

```diff
@@ -358,24 +358,12 @@ def get_grad(params_curr):
 
 from copy import copy
 
-
-def get_operations(qnode, params):
-    qnode.construct([params], {})
-    return qnode.tape.operations
-
-
 def get_overlap_tape(qnode, params1, params2):
-    op_forward = get_operations(qnode, params1)
-    op_inv = get_operations(qnode, params2)
-
-    with qml.tape.QuantumTape() as tape:
-        for op in op_forward:
-            qml.apply(op)
-        for op in reversed(op_inv):
-            qml.adjoint(copy(op))
-        qml.probs(wires=qnode.tape.wires.labels)
-    return tape
+    tape_forward = qml.workflow.construct_tape(qnode)(*params1)
+    tape_inv = qml.workflow.construct_tape(qnode)(*params2)
 
+    ops = tape_forward.operations + list(qml.adjoint(copy(op)) for op in reversed(tape_inv.operations))
```

Comment:
```suggestion
    ops = tape_forward.operations + list(qml.adjoint(op) for op in reversed(tape_inv.operations))
```
Sorry... just realized we don't need the copy.

Reply:
Done.